qa fixes and adaptions

bith3ad opened the pull request:
This PR is to switch to meta-arm and drop local trusted-firmware-a and fiptool recipes.

Commit:
meta-arm: add submodule

Add submodule to switch from custom trusted-firmware-a handling to the
upstream provided one.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: trusted-firmware-a: version bump to v2.10

Drop our downstream trusted-firmware-a.inc file and make use of the
meta-arm one to bump up the version. While on it replace our
tf-a-stm32mp_2.9.bb recipe with the meta-arm provided trusted-firmware-a
too. Furthermore add the "Upstream:" status to the patch to be Yocto
conform.

After the update the TF-A is printing:

| INFO:    PSCI Power Domain Map:
| INFO:      Domain Node : Level 1, parent_node 4294967295, State ON (0x0)
| INFO:      Domain Node : Level 0, parent_node 0, State ON (0x0)
| INFO:      CPU Node : MPID 0x0, parent_node 0, State ON (0x0)
| INFO:      CPU Node : MPID 0xffffffff, parent_node 0, State OFF (0x2)
| NOTICE:  CPU: STM32MP153C?? Rev.Z
| NOTICE:  Model: Linux Automation Test Automation Controller (TAC)
| NOTICE:  BL2: v2.10.4(release):lts-v2.10.4-dirty-20250401-1
| NOTICE:  BL2: Built : 16:23:33, Apr 26 2024
| NOTICE:  BL2: Booting BL32
| NOTICE:  SP_MIN: v2.10.4(release):lts-v2.10.4-dirty-20250401-1
| NOTICE:  SP_MIN: Built : 16:23:33, Apr 26 2024

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: drop local fiptool-native

Make use of the meta-arm fiptool-native recipe.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

jluebbe commented on .gitmodules:
```diff
@@ -26,3 +26,7 @@
+[submodule "meta-arm"]
+	path = meta-arm
+	url = https://git.yoctoproject.org/meta-arm
+	branch = scarthgap
```
Perhaps use https://github.com/jonmason/meta-arm so that dependabot can generate better changelogs?

Commit:
meta-lxatac-software: lxatac-core-image-base: add full blown ip command

Add support for the full-blown iprout2 ip command which is required for
configuring CAN interfaces.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-software: systemd: disable vconsole

Disable the virtual console support since it is not needed.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: enable dynamic tracing

This can be very useful for debugging and our hw does support it.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: set net default sched to fq-codel

fq_codel helps fight the network bufferbloat problem. It is believed to
be a good default with no tuning required for most workloads.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: disable rndis support

Disable the legacy rndis support which isn't used anyway.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: restrict dmesg to root

Only allow root to access dmesg since normal users don't need to read
it most the time.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: hardening the kernel

Enable some kernel hardening.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: enable hardlockup detector

This can be useful to detect hard to find issues early.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: disable unusual options

We don't use the isolcpus= param nor the legacy block autoload
mechanism, therefore disable it.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: disable unused VT options

We don't use the VT nor the PTYS support, therefore disable it.

Same is true for line-descipline drivers on-top of a serial device. The
line-descipline is replaced by serdev.

While on it, disable the rpmsg-tty driver support too since we don't use
the Cortex-M4 µC actually.

Disabling the DUMMY_CONSOLE is a side effect and is safe to do so.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: linux-lxatac: set squashfs accordingly

Invlove all CPUs to decompress squashfs files.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: barebox: sync defconfig

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: barebox: enable support for root=mmcblkXpN

Since our used kernel version supports deterministic MMC numbering we
can enable the feature to specify the root= device via mmcblkXpN.

Albeit the TAC supports only booting from eMMC making use of
root=mmcblkXpN is the preferrd solution, because it allows re-using the
same MMC image (same PARTUUID) for two differnt MMC devicess e.g. SD
and eMMC.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-software: lxatac-core-image-base: add dosfstools

Our kernel supports VFAT so the userspace should support fsck.vfat too.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

Commit:
meta-lxatac-bsp: barebox: remove depolying the barebox-dtbs

Actually we don't share the device-tree blobs any longer, therefore we can
drop deploying these to the DEPLOYDIR.

Signed-off-by: Marco Felsch <m.felsch@pengutronix.de>

hnez reviewed (approved):
I was most concerned with the witch to using the meta-arm provided `trusted-firmware-a`.
Mostly due to the clock setup done by the TF-A. But I have checked `/sys/kernel/debug/clk/clk_summary` before and after the change and did not see any significant changes, so I think we are good here.

The rest looks good to me as well.

bith3ad commented:
> I was most concerned with the witch to using the meta-arm provided `trusted-firmware-a`. Mostly due to the clock setup done by the TF-A. But I have checked `/sys/kernel/debug/clk/clk_summary` before and after the change and did not see any significant changes, so I think we are good here.
> 
> The rest looks good to me as well.

I did the same :-)
